hudi-common/src/main/java/org/apache/hudi/common/config/HoodieMetadataConfig.java

```diff
@@ -18,6 +18,9 @@
 
 package org.apache.hudi.common.config;
 
+import org.apache.hudi.common.engine.EngineType;
+import org.apache.hudi.exception.HoodieNotSupportedException;
+
 import javax.annotation.concurrent.Immutable;
 
 import java.io.File;
```

Comment:
Should `ENABLE` be set with `.noDefaultValue()`?  If not, engine-specific values are overwritten?

Reply:
This is what we do within getDefaultMetadataEnable(EngineType engineType).
```
case SPARK:
          return ENABLE.defaultValue();
```
probably we can leave it as is.

Reply:
Got it.  As discussed, functionality-wise this is fine.  Wondering if we can still set it with `.noDefaultValue()` instead of `.defaultValue(true)`, not to confuse developers on Java client.  I'll let you decide.

Reply:
I get it. but we might have to fix all tests to use the new constructor and so, don't want to fix this now. but will take it up sometime after 0.10.0.

Reply:
Makes sense.